test/test_tensor_creation_ops.py

```diff
@@ -940,7 +941,7 @@ def test_linspace(self, device, dtype):
                          atol=0, rtol=0)
 
         # Check for race condition (correctness when applied on a large tensor).
-        if dtype not in (torch.int8, torch.uint8, torch.int16, torch.half):
+        if dtype not in (torch.int8, torch.uint8, torch.int16, torch.half, torch.bfloat16) and not dtype.is_complex:
```

Comment:
Instead of skipping complex, would it make sense to take the `real` or `abs` for complex tensors and do the check?